### PR TITLE
Function debouncing utility

### PR DESCRIPTION
**Pull request by mike-north**
This [function debouncing utility](https://davidwalsh.name/javascript-debounce-function) is important for any webhook event handlers that observe multiple events. In cases where events like `pull_request` and `pull_request.edited` are fired in close succession. 

The two included test cases are the _minimal_ level of testing I'm comfortable with here, where we ensure that debouncing works effectively
- where multiple invocations happen in a single turn of the event loop
- where multiple invocations are scheduled into the event loop, but still occur within the "debounce time"

### CODE REVIEW

File: src/utils/debounce.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @param fn Function to debounce
+ * @param timeout how long (in ms) to debounce
+ */
+export function debounce<F extends (...args: any[]) => void>(
+  fn: F,
+  timeout: number,
+  state: WeakMap<Function, ReturnType<typeof setTimeout>> = defaultDebounceMap,
+): (...args: FArguments<F>) => Promise<void> {
+  return async function debounced(...args: FArguments<F>): Promise<void> {
```
Comment (chadhietala) [inline src/utils/debounce.ts]: What is actually being awaited on here?

Comment (mike-north) [reply src/utils/debounce.ts]: great point. I could just leave this as a non-async promise-returning function.

Comment (stefanpenner) [reply src/utils/debounce.ts]: @mike-north what does the promise returned fulfill or reject with?

Comment (mike-north) [reply src/utils/debounce.ts]: I was originally playing with the idea of debounce handling return values by returning a promise. There's no immediate need for this, so I've removed the return value entirely. Fix is in #24

**Pull request merged**